ui/cryptomaterial/dropdown.go

```diff
@@ -134,69 +152,78 @@ func (d *DropDown) Changed() bool {
 	return false
 }
 
-func (d *DropDown) layoutActiveIcon(gtx layout.Context, index int) D {
+func (d *DropDown) layoutActiveIcon(gtx C, index int) D {
 	var icon *Icon
 	if !d.isOpen {
 		icon = NewIcon(d.dropdownIcon)
 	} else if index == d.selectedIndex {
 		icon = NewIcon(d.navigationIcon)
 	}
 
-	return layout.E.Layout(gtx, func(gtx C) D {
-		if icon != nil {
-			icon.Color = d.theme.Color.Gray1
-			return icon.Layout(gtx, values.MarginPadding20)
-		}
-		return layout.Dimensions{}
-	})
+	if icon == nil {
+		return D{} // return early
+	}
+
+	if !d.isOpen || d.SelectedItemIconColor == nil {
+		icon.Color = d.theme.Color.Gray1
+	} else {
+		icon.Color = *d.SelectedItemIconColor
+	}
+
+	return icon.Layout(gtx, values.MarginPadding20)
 }
 
-func (d *DropDown) layoutOption(gtx layout.Context, itemIndex int) D {
+func (d *DropDown) layoutOption(gtx C, itemIndex int) D {
 	item := d.items[itemIndex]
 	radius := Radius(0)
 	clickable := item.clickable
 	if !d.isOpen {
 		radius = Radius(8)
 		clickable = d.clickable
+		clickable.Hoverable = d.MakeSelectedItemHoverable
 	}
 
 	padding := values.MarginPadding10
 	if item.Icon != nil {
 		padding = values.MarginPadding8
 	}
 
-	dropdownWidth := d.Width
-	dropdownItemWidth := dropdownWidth
-	if dropdownWidth <= 0 {
-		dropdownWidth = defaultDropdownWidth(gtx, d.revs)
-	}
-
 	return LinearLayout{
-		Width:     dropdownWidth,
+		Width:     MatchParent,
 		Height:    WrapContent,
 		Clickable: clickable,
 		Padding:   layout.UniformInset(padding),
 		Border:    Border{Radius: radius},
 	}.Layout(gtx,
 		layout.Rigid(func(gtx C) D {
 			if item.Icon == nil {
-				return layout.Dimensions{}
+				return D{}
 			}
 
-			dropdownItemWidth -= gtx.Dp(values.MarginPadding24) // account for the dropdown Icon
 			return item.Icon.Layout24dp(gtx)
 		}),
-		layout.Rigid(func(gtx C) D {
-			gtx.Constraints.Max.X = dropdownItemWidth - gtx.Dp(values.MarginPadding50) // give some space for the dropdown Icon
-			gtx.Constraints.Min.X = gtx.Constraints.Max.X
-			return layout.Inset{
-				Right: unit.Dp(5),
-				Left:  unit.Dp(5),
-			}.Layout(gtx, func(gtx C) D {
+		layout.Flexed(1, func(gtx C) D {
+			ll := LinearLayout{
+				Width:     MatchParent,
+				Height:    WrapContent,
+				Padding:   layout.Inset{Right: values.MarginPadding5},
+				Direction: layout.W,
+			}
+
+			if !d.isOpen {
+				ll.Direction = d.SelectedItemDirection
+			}
+
+			return ll.Layout2(gtx, func(gtx C) D {
+				if item.DisplayFn != nil {
+					return item.DisplayFn(gtx)
+				}
```

Comment:
If there's no alternative display, then this `d.SelectedItemDirection` is irrelevant. I wasn't even sure what it meant all along. Also, ll.Direction is irrelevant because there's just one item in the ll if `item.DisplayFn == nil`, if I'm not mistaken. In any case, let `item.DisplayFn` do all it needs to do, create an ll if it wants, set the direction it desires. Don't burden this method with those additional theatrics.